XcodeServerSDKTests/TestUtils.swift

```diff
@@ -56,4 +58,87 @@ extension XCTestCase {
         let configuration = bot.configuration
         return configuration
     }
+}
+
+// MARK: Exception assertions
+// Based on: https://forums.developer.apple.com/thread/5824
+extension XCTestCase {
+    /**
+    Replacement method for XCTAssertThrowsError which isn't currently supported.
+    
+    - parameter message: Message which should be displayed
+    - parameter file:    File in which assertion happened
+    - parameter line:    Line in which assertion happened
+    - parameter block:   Block of code against which assertion should be matched
+    */
+    func XCTempAssertThrowsError(message: String = "", file: String = __FILE__, line: UInt = __LINE__, _ block: () throws -> ()) {
+        do {
+            try block()
+            
+            let msg = (message == "") ? "Tested block did not throw error as expected." : message
+            XCTFail(msg, file: file, line: line)
+        } catch {}
+    }
+    
+    /**
+    Replacement method for XCTAssertThrowsSpecificError which isn't currently supported.
+    
+    - parameter kind:    ErrorType which is expected to be thrown from block
+    - parameter message: Message which should be displayed
+    - parameter file:    File in which assertion happened
+    - parameter line:    Line in which assertion happened
+    - parameter block:   Block of code against which assertion should be matched
+    */
+    func XCTempAssertThrowsSpecificError(kind: ErrorType, _ message: String = "", file: String = __FILE__, line: UInt = __LINE__, _ block: () throws -> ()) {
+        do {
+            try block()
+            
+            let msg = (message == "") ? "Tested block did not throw expected \(kind) error." : message
+            XCTFail(msg, file: file, line: line)
+        } catch let error as NSError {
+            let expected = kind as NSError
+            if ((error.domain != expected.domain) || (error.code != expected.code)) {
```

Comment:
I don't feel too confident with «downcasting» to `NSError`; feels like code smell.

Obviously nothing much we can  do since  is not providing a better way to; but I think we should mark this with a `TODO` or something and revisit later for when  supports this out of the box.